src/data/fintype.lean

```diff
@@ -221,6 +221,30 @@ instance (n : ℕ) : fintype (fin n) :=
 @[simp] theorem fintype.card_fin (n : ℕ) : fintype.card (fin n) = n :=
 list.length_fin_range n
 
+lemma fin.univ_succ (n : ℕ) :
+  (univ : finset (fin $ n+1)) = insert 0 (univ.image fin.succ) :=
+begin
+  ext m,
+  simp only [mem_univ, mem_insert, true_iff, mem_image, exists_prop],
+  exact fin.cases (or.inl rfl) (λ i, or.inr ⟨i, trivial, rfl⟩) m
+end
+
+theorem fin.prod_univ_succ [comm_monoid β] {n:ℕ} (f : fin n.succ → β) :
```

Comment:
Maybe add a comment why this does not work with `to_additive` (I guess this is because of the `0`?) and that the `to_additive` version will be registered in `fin.sum_univ_succ`.

Reply:
As far as I understand, it's because `+1` is used somewhere in the proof, and `to_additive` tries to replace it with `+0`.